parlai/scripts/train_model.py

```diff
@@ -242,9 +253,59 @@ def __init__(self, opt):
         self.saved = False
         self.valid_world = None
         self.opt = opt
+
+        # we may have been preempted, make sure we note that amount
+        self._preempted_epochs = 0.0
+        if (
+            opt.get('model_file') and
+            os.path.isfile(opt['model_file'] + trainstats_suffix)
```

Comment:
so is it _necessarily_ the case that we were preempted if there's a trainstats file? it looks like this is saved after every training run

Reply:
No, but if you finished training because of `--max-training-time` and then ran train_model again, you would immediately exit due to exceeding time.

Reply:
Ok, and I assume that is similar behavior to specifying `--num-epochs`